sort/bogosort.go

```diff
@@ -0,0 +1,32 @@
+// bogosort.go
+// description: Implementation of the bogosort algorithm
+// details:
+// A simple bogosort algorithm implementation. [Wikipedia](https://en.wikipedia.org/wiki/Bogosort)
+// author(s): [Focusucof](https://github.com/Focusucof)
+// see bogosort_test.go for tests
+
+package sort
+
+import (
+	"github.com/TheAlgorithms/Go/constraints"
+	"math/rand"
+	"sort"
+)
+
+// shuffles the given int array
+func shuffle(arr *[]int) {
+	rand.Shuffle(len(*arr), func(i, j int) {
+		(*arr)[i], (*arr)[j] = (*arr)[j], (*arr)[i] // swap values
+	})
+}
+
+// loops until array is sorted
+func BogoSort[T constraints.Ordered](arr []int) []int {
```

Comment:
```suggestion
func Bogo[T constraints.Ordered](arr []T) []T {
```

I think `Bogo` is the correct name here, sort is in the name of the package.

Also, you used `Ordered` `constraint` but didn't really use them anywhere in the algorithm. The function may seem generic to you right now, but it won't accept float slices even when you provide it with the float constraint. (you should check this yourself to better understand where you went wrong!)